rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java

```diff
@@ -147,43 +155,67 @@ private void getComponentsReports(boolean withLinkedReleases, boolean mailReques
-            throws TException {
+    
+    public String sourceBundleName;
+    private void getLicenseResourceBundleReports(String projectId, HttpServletRequest request,
+            HttpServletResponse response, User sw360User, String module) throws TException {
         try {
-            ByteBuffer buffer = null;
+            ByteBuffer buffer = sw360ReportService.downloadSourceCodeBundle(projectId, request, sw360User);
+            sourceBundleName = sw360ReportService.getSourceCodeBundleName(projectId, sw360User);
+            downloadExcelReport(false, response, sw360User, module, null, buffer);
+        } catch (Exception e) {
+            throw new TException(e.getMessage());
+        }
+    }
+
+    private void downloadExcelReport(boolean withLinkedReleases, HttpServletResponse response, User user, String module, String projectId, 
+            ByteBuffer buffer) throws TException {
+        try {
+            ByteBuffer buff = null;
             switch (module) {
-                case PROJECTS:
-                    buffer = sw360ReportService.getProjectBuffer(user, withLinkedReleases, projectId);
-                    break;
-                case COMPONENTS:
-                    buffer = sw360ReportService.getComponentBuffer(user, withLinkedReleases);
-                    break;
-                case LICENSES:
-                    buffer = sw360ReportService.getLicenseBuffer();
-                    break;
-                default:
-                    break;
+            case PROJECTS:
+                buff = sw360ReportService.getProjectBuffer(user, withLinkedReleases, projectId);
+                break;
+            case COMPONENTS:
+                buff = sw360ReportService.getComponentBuffer(user, withLinkedReleases);
+                break;
+            case LICENSES:
+                buff = sw360ReportService.getLicenseBuffer();
+                break;
+            case LICENSES_RESOURCE_BUNDLE:
+                buff = buffer;
+                break;
+            default:
+                break;
             }
-            if (null == buffer) {
+            if (null == buff) {
                 throw new TException("No data available for the user " + user.getEmail());
             }
-            response.setContentType(CONTENT_TYPE);
+
+            if (module.equalsIgnoreCase(LICENSES_RESOURCE_BUNDLE)) {
+                response.setContentType(ZIP_CONTENT_TYPE);
+            } else {
+                response.setContentType(CONTENT_TYPE);
+            }
+
             String fileName;
-            if(module.equals(LICENSES)) {
+            if (module.equals(LICENSES)) {
                 fileName = String.format("licenses-%s.xlsx", SW360Utils.getCreatedOn());
-            } else if(module.equals(PROJECTS)) {
+            } else if (module.equals(LICENSES_RESOURCE_BUNDLE)) {
+                fileName = sourceBundleName;
```

Comment:
The `projectId` and `sw360user` is available with the function. The function `sw360ReportService.getSourceCodeBundleName()` can be called to get the `fileName`.